etna/models/nn/nbeats/utils.py

```diff
@@ -0,0 +1,114 @@
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+
+import numpy as np
+
+from etna import SETTINGS
+
+if SETTINGS.torch_required:
+    import torch
+
+
+def _create_or_update(param: Optional[Dict], name: str, value: Any):
+    """Create new parameters dict or add field to existing one."""
+    if param is None:
+        param = {name: value}
+    else:
+        param[name] = value
+    return param
+
+
+def to_tensor(x: Any) -> "torch.Tensor":
+    """Convert data to tensor and put on default device.
+
+    Parameters
+    ----------
+    x:
+        Initial data for the tensor. Can be a list, tuple, NumPy ndarray, scalar, and other types.
+
+    Returns
+    -------
+    :
+        Input data as tensor.
+    """
+    return torch.tensor(x, dtype=torch.float32)
+
+
+def prepare_train_batch(
+    data: List[Dict[str, Any]],
+    input_size: int,
+    output_size: int,
+    window_sampling_limit: Optional[int] = None,
+    random_state: Optional[np.random.RandomState] = None,
+) -> Dict[str, Optional["torch.Tensor"]]:
+    """Prepare batch with training data."""
+    if random_state is None:
+        random_state = np.random.RandomState()
+
+    batch_size = len(data)
+
+    history = np.zeros((batch_size, input_size))
+    history_mask = np.zeros((batch_size, input_size))
+    target = np.zeros((batch_size, output_size))
+    target_mask = np.zeros((batch_size, output_size))
+
+    for i, part in enumerate(data):
+        series = part["history"]
+
+        if window_sampling_limit is not None:
+            lower_bound = max(1, len(series) - window_sampling_limit - 2)
```

Comment:
Why is -2?

Reply:
To ensure at least 1 point, when `window_sampling_limit = 0`

Reply:
It can seem a little bit strange to work with `window_sampling_limit=0`. Why not to work with negative values? I think that this value should represent some size, but now it represents it with non-obvious bias.